src/group_theory/order_of_element.lean

```diff
@@ -8,227 +8,222 @@ import group_theory.coset
+λ j, not_is_periodic_pt_of_pos_of_lt_minimal_period n0 h
+  ((is_periodic_pt_add_iff_nsmul_eq_zero x).mpr j)
 
-@[to_additive add_order_of_le_of_nsmul_eq_zero']
-lemma order_of_le_of_pow_eq_one' {m : ℕ} (h : m < order_of a) : ¬ (0 < m ∧ a ^ m = 1) :=
-begin
-  rw order_of at h,
-  split_ifs at h with hfin,
-  { exact nat.find_min hfin h },
-  { exact absurd h m.zero_le.not_lt }
-end
+@[to_additive nsmul_ne_zero_of_lt_add_order_of']
+lemma pow_eq_one_of_lt_order_of' (n0 : n ≠ 0) (h : n < order_of a) : a ^ n ≠ 1 :=
+λ j, not_is_periodic_pt_of_pos_of_lt_minimal_period n0 h
+  ((is_periodic_pt_mul_iff_pow_eq_one a).mpr j)
 
-lemma add_order_of_le_of_nsmul_eq_zero {n : ℕ} (hn : 0 < n) (h : n •ℕ x = 0) : add_order_of x ≤ n :=
-le_of_not_lt (mt add_order_of_le_of_nsmul_eq_zero' (by simp [hn, h]))
+lemma add_order_of_le_of_nsmul_eq_zero (hn : 0 < n) (h : n •ℕ x = 0) : add_order_of x ≤ n :=
+is_periodic_pt.minimal_period_le hn (by rwa is_periodic_pt_add_iff_nsmul_eq_zero)
 
 @[to_additive add_order_of_le_of_nsmul_eq_zero]
-lemma order_of_le_of_pow_eq_one {n : ℕ} (hn : 0 < n) (h : a ^ n = 1) : order_of a ≤ n :=
-le_of_not_lt (mt order_of_le_of_pow_eq_one' (by simp [hn, h]))
+lemma order_of_le_of_pow_eq_one (hn : 0 < n) (h : a ^ n = 1) : order_of a ≤ n :=
+is_periodic_pt.minimal_period_le hn (by rwa is_periodic_pt_mul_iff_pow_eq_one)
 
 @[simp] lemma order_of_one : order_of (1 : α) = 1 :=
 begin
-  apply le_antisymm,
-  { exact order_of_le_of_pow_eq_one (nat.one_pos) (pow_one 1) },
-  { exact nat.succ_le_of_lt ( order_of_pos' ⟨1, ⟨nat.one_pos, pow_one 1⟩⟩) }
+  rw [order_of, one_mul_eq_id],
+  exact minimal_period_id,
 end
 
 @[simp] lemma add_order_of_zero : add_order_of (0 : H) = 1 :=
-by simp [← order_of_of_add_eq_add_order_of]
+by simp only [←order_of_of_add_eq_add_order_of, order_of_one, of_add_zero]
 
 attribute [to_additive add_order_of_zero] order_of_one
 
 @[simp] lemma order_of_eq_one_iff : order_of a = 1 ↔ a = 1 :=
-⟨λ h, by conv_lhs { rw [← pow_one a, ← h, pow_order_of_eq_one] }, λ h, by simp [h]⟩
+by rw [order_of, is_fixed_point_iff_minimal_period_eq_one, is_fixed_pt, mul_one]
 
 @[simp] lemma add_order_of_eq_one_iff : add_order_of x = 1 ↔ x = 0 :=
 by simp [← order_of_of_add_eq_add_order_of]
 
 attribute [to_additive add_order_of_eq_one_iff] order_of_eq_one_iff
 
-lemma pow_eq_mod_order_of {n : ℕ} : a ^ n = a ^ (n % order_of a) :=
-calc a ^ n = a ^ (n % order_of a + order_of a * (n / order_of a)) : by rw [nat.mod_add_div]
-       ... = a ^ (n % order_of a) : by simp [pow_add, pow_mul, pow_order_of_eq_one]
-
-lemma nsmul_eq_mod_add_order_of {n : ℕ} : n •ℕ x = (n % add_order_of x) •ℕ x :=
+lemma nsmul_eq_mod_add_order_of : n •ℕ x = (n % add_order_of x) •ℕ x :=
 begin
-  apply multiplicative.of_add.injective,
-  rw [← order_of_of_add_eq_add_order_of, of_add_nsmul, of_add_nsmul, pow_eq_mod_order_of],
+  nth_rewrite 0 [← n.div_add_mod' (add_order_of x)],
+  rw [add_nsmul, mul_nsmul, add_order_of_nsmul_eq_zero, nsmul_zero, zero_add],
```

Comment:
It was a concious decision to change to a prove that used the analogous `minimal_period` statement. I wanted to make the two files look very much alike. If you have some good reasons not to do that, I would prefer the calc proof that was there before. Same applies to the multiplicative version.

Reply:
I do not mind to revert this proof and I just pushed this change!  By proof-irrelevance, once the proof is done, it does not matter what it is, in Lean!

Reply:
It could matter for speed or maintainability though.